From: Niko Tyni <ntyni@debian.org>
Subject: Document that XS_VERSION_BOOTCHECK may break if $VERSION is a long enough NV
Closes: 482139

Using a long floating point number as $VERSION can make
XS_VERSION_BOOTCHECK fail because of a loss of precision in the
stringification.

Upstream commit b017ed238008af06e2764e644dee23e72b0847ec


---
 pod/perlxs.pod |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/pod/perlxs.pod b/pod/perlxs.pod
index facfcf9..937883b 100644
--- a/pod/perlxs.pod
+++ b/pod/perlxs.pod
@@ -1212,6 +1212,12 @@ To disable version checking:
 
     VERSIONCHECK: DISABLE
 
+Note that if the version of the PM module is an NV (a floating point
+number), it will be stringified with a possible loss of precision
+(currently chopping to nine decimal places) so that it may not match
+the version of the XS module anymore. Quoting the $VERSION declaration
+to make it a string is recommended if long version numbers are used.
+
 =head2 The PROTOTYPES: Keyword
 
 The PROTOTYPES: keyword corresponds to B<xsubpp>'s C<-prototypes> and
-- 
tg: (71b2123..) fixes/doc-xs-versioncheck (depends on: upstream)
